### PR TITLE
Update WindowsBuild.md

### PR DESCRIPTION
**Pull request by compnerd**
Enhance the documentation to cover building SQLite, llbuild, and swift-package-manager.  This Is not fully ready, but this allows others to start collaborating on the work necessary to complete the work needed to support Windows in swift-package-manager.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @jmittert @jakepetroules - they have been working on llbuild
CC: @aciidb0mb3r @ddunbar @dmbryson - they would care about s-p-m

File: docs/WindowsBuild.md [hunk 1]
```diff
@@ -350,14 +362,55 @@ cmake -G Ninja^
...
+```cmd
+path S:\b\llbuild;%PATH%
+```
+
+### 21. Build swift-package-manager
+
+```cmd
+md S:\b\spm
+cd S:\b\spm
+C:\Python27\python.exe S:\swift-package-manager\Utilities\bootstrap --foundation S:\b\foundation --libdisptch-build-dir S:\b\libdispatch --libdispatch-source-dir S:\swift-corelibs-libdispatch
```
Comment (compnerd) [inline docs/WindowsBuild.md]: Typo: `libdisptch` -> `libdispatch`

File: docs/WindowsBuild.md [hunk 2]
```diff
@@ -41,6 +41,8 @@ it provides some of the needed headers and libraries.
 1. Clone `apple/swift-corelibs-foundation` into a folder named `swift-corelibs-foundation`
 1. Clone `apple/swift-corelibs-xctest` into a folder name `swift-corelibs-xctest`
 1. Clone `apple/swift-lldb` into a folder named `lldb`
+1. Clone `apple/swift-llbuild` into a folder named `swift-llbuild`
```
Comment (compnerd) [inline docs/WindowsBuild.md]: This needs to be checked out to `llbuild`.

File: docs/WindowsBuild.md [hunk 3]
```diff
@@ -41,6 +41,8 @@ it provides some of the needed headers and libraries.
...
+1. Clone `apple/swift-package-manager` into a folder named `swift-package-manager`
```
Comment (shahmishal) [inline docs/WindowsBuild.md]: Not sure, if this true on Windows but on macOS it's configured to swiftpm.

Comment (gwynne) [reply docs/WindowsBuild.md]: AFAIK the only repos for which the name actually _matters_ are `llvm`, `clang`, and `compiler-rt`; the rest just need to be used consistently, at least as things currently stand.

Comment (compnerd) [reply docs/WindowsBuild.md]: As I've been working on the s-p-m port, I've learnt that `llbuild` is also an odd apple that forks.

**Pull request closed without merging**